File: 1. Two Sum/index2.js
/**
 * @description https://leetcode.com/problems/two-sum/
 * @type self
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  var map = {};
  for (let i = 0; i < nums.length; i++) {
    if (typeof map[target - nums[i]] !== 'undefined') {
      return [i, map[target - nums[i]]];
    }
    map[nums[i]] = i;
  }
  return [];
};

File: 509. Fibonacci Number/index2.java
/**
 * https://leetcode.com/problems/fibonacci-number/
 * self
 */
class Solution {
  public int fib(int n) {
    if (n == 0)
      return 0;
    if (n == 1 || n == 2)
      return 1;
    int prev = 1, cur = 1;
    for (int i = 3; i <= n; i++) {
      int sum = prev + cur;
      prev = cur;
      cur = sum;
    }
    return cur;
  }
}